plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1587,6 +1594,10 @@ void producer_plugin_impl::plugin_startup() {
          _irreversible_block_time = fc::time_point::maximum();
       }
 
+      if (!_is_savanna_active && irreversible_mode() && chain_plug->accept_transactions()) {
+         wlog("Accepting speculative transaction execution not recommended in read-mode=irreversible");
```

Comment:
If this check is needed, add something indicating nodeos is in Legacy.

Reply:
https://github.com/AntelopeIO/spring/pull/1036/commits/792f4778182d9c80ac7cfe2995b05e6d060b8776